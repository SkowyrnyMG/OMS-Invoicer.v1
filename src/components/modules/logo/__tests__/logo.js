import React from 'react';

import {
  renderWithReduxRouter,
  snapShotTest,
  screen,
  userEvent,
} from 'utils/tests/test-helper';
import Logo from '../logo';

describe('Logo', () => {
  test('should display in the document with correct styles', () => {
    snapShotTest(<Logo />);
  });

  test('should redirect to home page after logo click', () => {
    const { history } = renderWithReduxRouter(
      <Logo />,
      {},
      { route: '/login' },
    );

    const LogoNode = screen.getByRole('link', { name: 'logo.svg Invoicer.v1' });
    const initialPath = history.location.pathname;

    userEvent.click(LogoNode);

    const finalPath = history.location.pathname;

    expect(initialPath).not.toBe(finalPath);
    expect(finalPath).toBe('/');
  });
});
