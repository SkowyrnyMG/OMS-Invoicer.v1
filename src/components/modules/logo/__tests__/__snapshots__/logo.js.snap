// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should display in the document with correct styles 1`] = `
.c0 {
  display: inline-block;
  padding: 1rem 2rem;
  margin-top: 0;
  font-weight: 700;
  color: #000 !important;
  border-left: 1px solid #F0F0FD;
  background: none;
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
  border: none;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 20px;
  font-weight: 700;
}

.c2 {
  width: 6rem;
}

@media (max-width:1300px) {
  .c3 {
    display: none;
  }
}

<a
  class="c0"
  href="/"
  linktype=""
>
  <span
    class="c1"
  >
    <svg
      class="c2"
    >
      logo.svg
    </svg>
    <span
      class="c3"
    >
      Invoicer.v1
    </span>
  </span>
</a>
`;
